OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -160,12 +175,14 @@ void IBotTick.BotTick(IBot bot)
 
 		internal Actor FindClosestEnemy(WPos pos)
 		{
-			return World.Actors.Where(IsEnemyUnit).ClosestTo(pos);
+			var units = World.Actors.Where(IsEnemyUnit);
+			return units.Where(IsNotHiddenUnit).ClosestTo(pos) ?? units.ClosestTo(pos);
 		}
 
 		internal Actor FindClosestEnemy(WPos pos, WDist radius)
 		{
-			return World.FindActorsInCircle(pos, radius).Where(IsEnemyUnit).ClosestTo(pos);
+			var units = World.FindActorsInCircle(pos, radius).Where(IsEnemyUnit);
+			return units.Where(IsNotHiddenUnit).ClosestTo(pos);
```

Comment:
This is only a question for now, not a change request (yet): Is there a reason why this doesn't do the same ` ?? units.ClosestTo(pos)` as the overload above?

Reply:
Because find actors in `FindClosestEnemy(WPos pos, WDist radius)` is often used for target scanning within an area. If we apply `?? units.ClosestTo(pos)` here then AI will try to attack it (but stucked because it cannot be attacked when cloaked). 

`FindClosestEnemy(WPos pos)` is often used for approaching target for AI squad. So `?? units.ClosestTo(pos)` here means if AI really cannot find anything that not cloaked, they will finally try to approach the cloacked one.

Reply:
In that case it should be possible to combine the two `Where()` though, since both need to be checked anyway. Then we also wouldn't need the `var` and could just return
`World.FindActorsInCircle(pos, radius).Where(a => IsEnemyUnit(a) && IsNotHiddenUnit(a)).ClosestTo(pos)`, which should be a bit more performance-efficient.

Reply:
Fixed